File: ai-job-server/src/main/java/com/aijob/server/entity/JobMatch.java
package com.aijob.server.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("job_match")
public class JobMatch {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long resumeId;

    private String jobName;

    private String companyName;

    private String jobDescription;

    private Integer matchScore;

    /** JSON 数组字符串 */
    private String matchedSkills;

    /** JSON 数组字符串 */
    private String missingSkills;

    /** JSON 数组字符串 */
    private String advantages;

    /** JSON 数组字符串 */
    private String risks;

    /** JSON 数组字符串 */
    private String suggestions;

    private String summary;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: ai-job-web/src/types/jobMatch.ts
import type { Resume } from '@/types/resume'

export interface JobMatchRequest {
  resumeId: number
  jobName: string
  companyName?: string
  jobDescription: string
}

export interface JobMatch {
  id: number
  userId: number
  resumeId: number
  jobName: string
  companyName?: string
  jobDescription: string
  matchScore: number
  matchedSkills: string[] | string
  missingSkills: string[] | string
  advantages: string[] | string
  risks: string[] | string
  suggestions: string[] | string
  summary: string
  createdAt?: string
  updatedAt?: string
}

/** 解析 JSON 数组字符串或直接数组 */
export function parseJsonList(value: string[] | string | null | undefined): string[] {
  if (value == null) return []
  if (Array.isArray(value)) {
    return value.map((item) => String(item).trim()).filter(Boolean)
  }
  const text = String(value).trim()
  if (!text) return []
  try {
    const parsed = JSON.parse(text)
    if (Array.isArray(parsed)) {
      return parsed.map((item) => String(item).trim()).filter(Boolean)
    }
  } catch {
    // 兼容逗号分隔
    return text.split(',').map((item) => item.trim()).filter(Boolean)
  }
  return []
}

export type { Resume }
